torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -539,17 +538,17 @@ class ShapePropagator {
         return;
       }
       case aten::_unwrap_optional: {
-	// if we have None as input, we need to leave the output alone
+        // if we have None as input, we need to leave the output alone
         auto input_ivalue = toIValue(node->input());
         if (input_ivalue && input_ivalue->isNone()) {
           return;
         }
-	if(auto ot = node->input()->type()->cast<OptionalType>()) {
-	  node->output()->setType(ot->getElementType());
-	} else {
-	  node->output()->setType(node->input()->type());
-	}
-	return;
+        if (auto ot = node->input()->type()->cast<OptionalType>()) {
+          node->output()->setType(ot->getElementType());
+        } else if (!node->input()->type()->isSubtypeOf(NoneType::get())) {
```

Comment:
Why?

Reply:
During analysis, we may have None going in when we specialize the graph. This will be in a branch that is later pruned / not executed, but it'll screw the analysis.
I'll add this as a code comment.

Reply:
Why would it screw the analysis? If we know it will deleted later, why does it matter?

Reply:
Well, the code following the unwrap will need a containing type. So we'll hit asserts if we continue with None.

Reply:
So the None part is gone now, and below, Zach has comments on the same place.